apps/api-extractor/src/analyzer/ExportAnalyzer.ts

```diff
@@ -266,29 +262,23 @@ export class ExportAnalyzer {
   /**
    * Returns true if the module specifier refers to an external package.  Ignores packages listed in the
    * "bundledPackages" setting from the api-extractor.json config file.
-   *
-   * @remarks
-   * Examples:
-   *
-   * - NO:  `./file1`
-   * - YES: `library1/path/path`
-   * - YES: `@my-scope/my-package`
    */
-  private _isExternalModulePath(moduleSpecifier: string): boolean {
-    if (ts.isExternalModuleNameRelative(moduleSpecifier)) {
-      return false;
-    }
+  private _isExternalModulePath(
+    importOrExportDeclaration: ts.ImportDeclaration | ts.ExportDeclaration | ts.ImportTypeNode,
+    moduleSpecifier: string
+  ): boolean {
+    const resolvedModule: ts.ResolvedModuleFull = this._getResolvedModule(
+      importOrExportDeclaration,
+      moduleSpecifier
+    );
 
-    const match: RegExpExecArray | null = ExportAnalyzer._modulePathRegExp.exec(moduleSpecifier);
-    if (match) {
-      // Extract "@my-scope/my-package" from "@my-scope/my-package/path/module"
-      const packageName: string = match[1];
-      if (this._bundledPackageNames.has(packageName)) {
-        return false;
-      }
+    // Either something like `jquery` or `@microsoft/api-extractor`.
+    const packageName: string | undefined = resolvedModule.packageId?.name;
+    if (packageName !== undefined && this._bundledPackageNames.has(packageName)) {
+      return false;
     }
 
-    return true;
+    return !!resolvedModule.isExternalLibraryImport;
```

Comment:
It's unclear to me when `isExternalLibraryImport` is `undefined`... but it's currently typed as `boolean | undefined`. I couldn't repro a scenario where this was `undefined`.

Reply:
Looking at the compiler source code, it seems to be undefined when the resolution failed, so it's unknown whether the import was external or not: (?)

[src/compiler/moduleNameResolver.ts](https://github.com/microsoft/TypeScript/blob/41aca7c337dbe4a87ea23f8a91efd39e0ae81a1e/src/compiler/moduleNameResolver.ts#L1313)
```ts
        const result = forEach(extensions, ext => tryResolve(ext));
        return createResolvedModuleWithFailedLookupLocations(result?.value?.resolved, result?.value?.isExternalLibraryImport, failedLookupLocations, state.resultFromCache);
```

Reply:
Based on that, I think we might prefer the opposite default.  If we accidentally treat something as external, it will get skipped by the analysis. Whereas if accidentally treat something as internal, API Extractor can get lost in the woods trying to analyze third party .d.ts files.

```ts
if (resolvedModule.isExternalLibraryImport === undefined) {
  // If the compiler doesn't know whether the declaration is external,
  // assume it's external
  return true;
}

return resolvedModule.isExternalLibraryImport;
```

Reply:
We could also throw an `InternalError`, similarly to how we throw when `getResolvedModule` returns `undefined`?

Reply:
Actually yes, let's start with `InternalError`.  If this is happening in normal usage, I'd rather investigate an `InternalError` callstack than a mysterious malfunction with no idea where to start.